docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -0,0 +1,475 @@
+`ML_TRAIN_LAYER_TYPE_FC`    |                    |               | Fully connected layer
+&#xfeff;                    | unit               | (integer)     | number of outputs
+
+### Optimizer
+
+Optimizer determines how to update model parameters according to loss from prediction.
+Currently, Stochastic Gradient Descent optimizer and Adam optimizer are supported:
+
+```c
+// Create an optimizer
+ml_train_optimizer_h optimizer;
+ml_train_optimizer_create(&optimizer, ML_TRAIN_OPTIMIZER_TYPE_SGD);
+
+// Configure a optimizer
+ml_train_optimizer_set_property(optimizer, "learning_rate=0.001", NULL);
+
+// Set optimizer to the model
+// don't need to destroy optimizer after setting optimizer
+// since ownership is transferred to the model.
+ml_train_model_set_optimizer(model, optimizer);
+```
+
+Following are the available properties for each optimizer type:
+
+Type | Key | value | Description
+---------- | --- | ----- | -----------
+(Universal properties)                   |                    |               | Universal properties that applies to every layer
+&#xfeff;                            | learning_rate      | (float)       | Initial learning rate for the optimizer
+`ML_TRAIN_OPTIMIZER_TYPE_SGD`       |                    |               | Stochastic Gradient Descent optimizer
+`ML_TRAIN_OPTIMIZER_TYPE_ADAM`      |                    |               | Adam optimizer
+&#xfeff;                            | decay_steps        | (float)       | Decay steps
+&#xfeff;                            | decay_rate         | (float)       | Decay rate
+&#xfeff;                            | beta1              | (float)       | beta1 coefficient for adam
+&#xfeff;                            | beta2              | (float)       | beta2 coefficient for adam
+&#xfeff;                            | epsilon            | (float)       | epsilon coefficient for adam
+
+### Dataset
+
+Dataset is in charge of feeding data into the model.
+The dataset can either be created from a callback function or created from a file.
+For more information, see [configure the model](#configure-the-model) section.
+
+Following code is example of handling dataset:
+
+```c
+// Create dataset
+ml_train_dataset_h dataset;
+ml_train_dataset_create_with_generator(&dataset, generator_train_cb, generator_valid_cb, generator_test_cb);
+
+// configure dataset
+ml_train_dataset_set_property(dataset, "buffer_size=100", NULL);
+
+// after setting a dataset to model,
+// you do not need to destroy dataset since ownership is transferred to the model.
+ml_train_model_set_dataset(model, dataset);
+```
+
+## Construct a model
+
+A model can be constructed with `ml_train_model_construct()`.
+If you have a file that describes the model, the file can be used to construct initially with `ml_train_model_construct_with_file()`.
+Even if the model is constructed from a file, switching, modifying, or setting a component is possible only until you compile with `ml_train_model_compile()`.
```

Comment:
In my opinion usage of "only" is fine here as it is emphasizing.

Reply:
My concern was that the "only" is emphasizing the sentence in negative tone. 
In my understanding the tone of the original sentence was positive, and it is changed to negative tone now.
If I understood something wrong, please let me know.

Reply:
As I asked to @surbhig12 I think removing "only" is better. I hope @zhoonit also check these comments.

Reply:
"only" removed.